paradox-nakadi-consumer-boot/src/main/java/de/zalando/paradox/nakadi/consumer/boot/FailedEventReplayer.java

```diff
@@ -0,0 +1,88 @@
+package de.zalando.paradox.nakadi.consumer.boot;
+
+import static java.util.Objects.requireNonNull;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.stereotype.Component;
+
+import de.zalando.paradox.nakadi.consumer.boot.components.EventReceiverRegistry;
+import de.zalando.paradox.nakadi.consumer.boot.components.EventTypeConsumer;
+import de.zalando.paradox.nakadi.consumer.boot.components.FailedEventSourceMap;
+import de.zalando.paradox.nakadi.consumer.core.EventHandler;
+import de.zalando.paradox.nakadi.consumer.core.FailedEventSource;
+import de.zalando.paradox.nakadi.consumer.core.domain.EventTypePartition;
+import de.zalando.paradox.nakadi.consumer.core.domain.FailedEvent;
+
+@Component
+public class FailedEventReplayer {
+
+    private final EventReceiverRegistry eventReceiverRegistry;
+
+    private final FailedEventSourceMap failedEventSourceMap;
+
+    private final ReplayHandler replayHandler;
+
+    private static final String EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE = "Event source name is not available.";
+
+    @Autowired
+    public FailedEventReplayer(final EventReceiverRegistry eventReceiverRegistry,
```

Comment:
It's unclear what `FailedEventSourceMap` is storing. The map is mutable and it is possible to construct a map with invalid state, i.e., wrong key value mapping.

You can  autowire `List<FailedEventSource>` and construct the map in this class.

`@Autowired` Javadoc
> In case of a java.util.Collection or java.util.Map dependency type, the container will autowire all beans matching the declared value type. In case of a Map, the keys must be declared as type String and will be resolved to the corresponding bean names.

This will move into `FailedEventReplayer`.
```
    @Bean
    public FailedEventSourceMap failedEventSourceMap() {

        final Map<String, FailedEventSource> failedEventSourceMap = applicationContext.getBeansOfType(
                FailedEventSource.class);

        if (null != failedEventSourceMap) {
            return new FailedEventSourceMap(failedEventSourceMap.entrySet().stream().collect(Collectors.toMap(k ->
                                k.getValue().getEventSourceName(),
                            Map.Entry::getValue)));
        } else {
            return new FailedEventSourceMap();
        }
    }
```

Also, `FailedEventReplayer` is a `@Component` but you have 
```
    @Bean
    public FailedEventReplayer failedEventHandler(final EventReceiverRegistry eventReceiverRegistry,
            final FailedEventSourceMap failedEventSourceMap) {
        return new FailedEventReplayer(eventReceiverRegistry, failedEventSourceMap, new ReplayHandler());
    }

```